config.yml

```diff
@@ -1,21 +1,134 @@
+# kick_thresholds field ignores non-numerical characters and only reads three numbers separated, 
+#   non-numericals are there only to make configuration file more readable.
+#   Use 0 as an amount of seconds when you don't want to kick people.
 
 kick_thresholds:
-  - When 81 to 100 players online kick after 60 seconds
-  - When 51 to 80  players online kick after 1000 seconds
-  - When 2  to 50  players online kick after 20 seconds
-  - When 1  to 1  players online kick after 40 seconds  
-  
-  
-Comment: warnings filed reads the first number, the rest is text to be sent to a player
+  - When 81 to 500 players online kick after 60 seconds
+  - When 51 to 80  players online kick after 120 seconds
+  - When 2  to 50  players online kick after 240 seconds
+  - When 1  to 1  players online kick after 3600 seconds  
   
+bot_detector:
+  # Once the TPS get below acceptable_TPS, the BotDetector starts culling potential lag causer.
+  acceptable_TPS: 17.5
+  # critical_TPS_Change is the minimal change in TPS that needs to happen after kicking a player, 
+  #   so he is marked as someone causing lag and his name is logged.
+  critical_TPS_Change: 0.5
+  # kicking_frequency is how often a player gets kicked and the TPS change monitored afterwards 
+  #   in ticks
+  kicking_frequency: 1200
+  # Whether players get banned for a long time/ until the tick is high again
+  long_bans: false
+  # Length of long ban
+  ban_length: 21600000
+  # Max locations per player to store while measuring movement patterns
+  max_locations: 5
+  # Max suspects to consider per Detector round
+  max_suspects: 10
+  # Baseline movement (in MC blocks) to be considered for bot detection
+  min_baseline: 512
+  # Maximum scan rounds to "ignore" a player who was previously cleared as a lag source
+  max_reprieve: 20
+  # Scan radius in chunks around location, to check for lag causes
+  scan_radius: 2
+  # Configure the threshold and individual contribution of bound types
+  bounds:
+    relaxation_factor: 1.05
+    threshold: 2.5
+    contained: 1.5
+    contained_exclude_y: 1.5
+    volume_similar: 1.0
+    surface_similar: 1.0
+    nearly_contained: 0.75
+    nearly_contained_exclude_y: 0.75
+
+lag_scanner:
+  # Length of time to cache individual chunk results
+  cache_timeout: 900000
+  # Summation threshold of entity and tile costs for lag flag
+  lag_threshold: 750
+  # Per tickable block cost 
+  tick_block:
+    BANNER: 1
+    BEACON: 5
+    BREWING_STAND: 1
+    MOB_SPAWNER: 3
+    DROPPER: 3
+    JUKEBOX: 1
+    NOTE_BLOCK: 1
+    SIGN: 1
+    HOPPER: 5
+    DISPENSER: 3
+    FURNACE: 3
+    BURNING_FURNACE: 4
+    CHEST: 1
+    ACTIVATOR_RAIL: 1
+    DETECTOR_RAIL: 1
+    DAYLIGHT_DETECTOR: 1
+    DAYLIGHT_DETECTOR_INVERTED: 1
+    DIODE: 1
+    DIODE_BLOCK_OFF: 1
+    DIODE_BLOCK_ON: 1
+    PISTON_BASE: 2
+    PISTON_STICKY_BASE: 2
+    POWERED_RAIL: 1
+    REDSTONE_COMPARATOR: 1
+    REDSTONE_COMPARATOR_OFF: 1

```

Comment:
Because of the cart lag and fully powered carts stopping on the tracks, it got common practice to use powered rails only for any sort of cart collection system. I know enough places that have 1k+ powered rails, those machines usually aren't used that often, but right now it would ban people just for walking close to those monsters. Do powered rails cause lag if they are not powering any carts?
Also what brought up the thought for me was, that for example Thule has about 1k+ comparators below the city, which are intended to be used once every few months. People would pretty much get banned there all the time for standing in the middle of the town, while actually no redstone is powered and no lag is being caused.  
Independent from this, I think that many people aren't really aware how much lag they cause with tile entities (maybe that includes what I just tried to defend) and when we release this, we should tell people to reduce the amount of tile entities around them to reduce their chance of being kicked, this might bring a performance improvement on its own.